crates/swift-bridge-ir/src/codegen/generate_rust_tokens.rs

```diff
@@ -250,6 +255,29 @@ impl ToTokens for SwiftBridgeModule {
                         }
                     };
                 }
+                TypeDeclaration::CustomResult(custom_result) => {
```

Comment:
As mentioned above, I don't think that we need a separate `CustomResult` type. We can just use the `BuiltInResult`
Where the `BuiltIn` just means that we're using the Result from the stdlib. Then for certain ok/err combinations we emit some custom Rust and C code for the FFI representation.

This way any code that we write in the future that cares about results doesn't need to check both for built in results and custom results.